src/handle_wrap.h

```diff
@@ -88,15 +88,18 @@ class HandleWrap : public AsyncWrap {
 
   void MarkAsInitialized();
   void MarkAsUninitialized();
+  void MarkAsClosing();
+  bool IsInitialized() { return state_ == kInitialized; }
 
   inline bool IsHandleClosing() const {
     return state_ == kClosing || state_ == kClosed;
   }
 
+  static void OnClose(uv_handle_t* handle);
```

Comment:
changes to underlying streams make me nervous, is this needed by bob? by js_udp_wrap? just the src/quic/... ?

Reply:
No, these are needed because of the sync timing differences in the cleanup of QUIC objects. Nothing to do with the bob streams. This can definitely be separated out but nothing else in core would use it until QUIC landed.